src/backend/vulkanll/src/factory.rs

```diff
@@ -178,128 +278,345 @@ impl core::Factory<R> for Factory {
+                p_next: ptr::null(),
+                flags: vk::PipelineDynamicStateCreateFlags::empty(),
+                dynamic_state_count: dynamic_states.len() as u32,
+                p_dynamic_states: dynamic_states.as_ptr(),
+            });
+
+            Ok(vk::GraphicsPipelineCreateInfo {
+                s_type: vk::StructureType::GraphicsPipelineCreateInfo,
+                p_next: ptr::null(),
+                flags: vk::PipelineCreateFlags::empty(),
+                stage_count: info_stages.last().unwrap().len() as u32,
+                p_stages: info_stages.last().unwrap().as_ptr(),
+                p_vertex_input_state: info_vertex_input_states.last().unwrap(),
+                p_input_assembly_state: info_input_assembly_states.last().unwrap(),
+                p_rasterization_state: info_rasterization_states.last().unwrap(),
+                p_tessellation_state: if desc.shader_entries.hull_shader.is_some() &&
+                                         desc.shader_entries.domain_shader.is_some()
+                    { info_tessellation_states.last().unwrap() } else { ptr::null() },
+                p_viewport_state: info_viewport_states.last().unwrap(),
+                p_multisample_state: info_multisample_states.last().unwrap(),
+                p_depth_stencil_state: info_depth_stencil_states.last().unwrap(),
+                p_color_blend_state: info_color_blend_states.last().unwrap(),
+                p_dynamic_state: info_dynamic_states.last().unwrap(),
                 layout: signature.layout,
                 render_pass: subpass.main_pass.inner,
                 subpass: subpass.index as u32,
                 base_pipeline_handle: vk::Pipeline::null(),
                 base_pipeline_index: -1,
             })
         }).collect::<Vec<_>>();
+
+        let valid_infos = infos.iter().filter_map(|info| info.clone().ok()).collect::<Vec<_>>();
         
         // TODO: create the pipelines!
+        let pipelines = if valid_infos.is_empty() {
+            Ok(Vec::new())
+        } else {
+            unsafe {
+                self.inner.0.create_graphics_pipelines(
+                    vk::PipelineCache::null(),
+                    &valid_infos,
+                    None,
+                )
+            }
+        };
+
+        match pipelines {
+            Ok(pipelines) => {
+                let mut pipelines = pipelines.iter();
+                infos.iter().map(|ref info| {
+                    match **info {
+                        Ok(_) => {
+                            let pipeline = native::GraphicsPipeline {
+                                pipeline: *pipelines.next().unwrap(),
+                            };
+                            Ok(pipeline)
+                        }
+                        Err(ref err) => Err(err.clone()),
+                    }
+                }).collect::<Vec<_>>()
+            },
+            Err(err) => {
```

Comment:
On error I should check each returned handle for `VK_NULL_HANDLE` and return error on null handle. Currently returning error if *one* pipeline creation fails, which isn't really useful. I'm going to fix this!